File: Examples/VisualBasic/WorkingWithProjects/WorkingWithExtendedAttributes/WriteExtendedAttributes.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Xml
Imports Aspose.Tasks

Namespace WorkingWithProjects.WorkingWithExtendedAttributes
    Public Class WriteExtendedAttributes
        Public Shared Sub Run()
            ' ExStart: ExtendedAttributes
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            Dim prj As New Project(dataDir & "ExtendedAttributes.mpp")
            Dim eads As ExtendedAttributeDefinitionCollection = prj.ExtendedAttributes

            ' Create extended attribute definition
            Dim ead As New ExtendedAttributeDefinition()
            ead.FieldId = (CInt(Fix(ExtendedAttributeTask.Start7))).ToString()
            ead.FieldName = "Start7"
            eads.Add(ead)

            ' Get zero index task
            Dim tsk As Task = prj.RootTask.Children.GetById(1)
            Dim eas As ExtendedAttributeCollection = tsk.ExtendedAttributes

            ' Add extended attribute
            Dim dateTimeFormat As String = "yyyy-MM-ddTHH:mm:ss"
            Dim ea As New ExtendedAttribute()
            ea.FieldId = ead.FieldId
            ea.Value = XmlConvert.ToString(DateTime.Now, dateTimeFormat)
            eas.Add(ea)
            ' ExEnd: ExtendedAttributes
        End Sub
    End Class
End Namespace

File: Examples/CSharp/WorkingWithProjects/WorkingWithExtendedAttributes/WriteExtendedAttributes.cs
namespace Aspose.Tasks.Examples.CSharp.WorkingWithProjects.WorkingWithExtendedAttributes
{
    using System;
    using Saving;

    internal class WriteExtendedAttributes
    {
        public static void Run()
        {
            // The path to the documents directory.
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName);

            //ExStart:ExtendedAttributes
            //ExFor: ExtendedAttributeDefinition.CreateTaskDefinition(CustomFieldType,ExtendedAttributeTask,String)
            //ExFor: Task.ExtendedAttributes
            //ExSummary: Shows how to write extended attributes.
            var project = new Project(dataDir + "ExtendedAttributes.mpp");

            // Create extended attribute definition
            var definition = ExtendedAttributeDefinition.CreateTaskDefinition(CustomFieldType.Start, ExtendedAttributeTask.Start7, "Start 7");
            project.ExtendedAttributes.Add(definition);

            // Get zero index task
            var task = project.RootTask.Children.GetById(1);

            // Add extended attribute
            var attribute = definition.CreateExtendedAttribute();
            attribute.DateValue = DateTime.Now;

            // Also the following short syntax can be used: ExtendedAttribute attribute = attributeDefinition.CreateExtendedAttribute(DateTime.Now);
            task.ExtendedAttributes.Add(attribute);
            project.Save(dataDir + "WriteExtendedAttributes_out.mpp", SaveFileFormat.MPP);
            //ExEnd:ExtendedAttributes
        }
    }
}